packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -496,177 +496,187 @@ export default class TableResize implements EditorPlugin {
 
-            let currentBorder: number = parseFloat(
-                this.currentTable.getAttribute(
-                    this.isRTL ? 'currentLeftBorder' : 'currentRightBorder'
-                )
-            );
-            const tableBottomBorder: number = parseFloat(
-                this.currentTable.getAttribute('currentBottomBorder')
-            );
-            const ratioX =
-                1.0 +
-                (this.isRTL
-                    ? (currentBorder - e.pageX) / (rect.right - currentBorder)
-                    : (e.pageX - currentBorder) / (currentBorder - rect.left));
-            const ratioY = 1.0 + (e.pageY - tableBottomBorder) / (tableBottomBorder - rect.top);
-
-            const shouldResizeX = Math.abs(ratioX - 1.0) > 1e-3;
-            const shouldResizeY = Math.abs(ratioY - 1.0) > 1e-3;
-            if (shouldResizeX || shouldResizeY) {
-                for (let i = 0; i < vtable.cells.length; i++) {
-                    for (let j = 0; j < vtable.cells[i].length; j++) {
-                        const cell = vtable.cells[i][j];
-                        if (cell.td) {
-                            if (shouldResizeX) {
-                                const originalWidth: number = cell.td.style.width
+        if (!this.isReadyForResizingTable) {
+            preProcessTable(vtable.table, ResizeState.Both);
+            this.isReadyForResizingTable = true;
+        }
+        let rect = normalizeRect(vtable.table.getBoundingClientRect());
+        let currentBorder: number = parseFloat(
+            vtable.table.getAttribute(this.isRTL ? 'currentLeftBorder' : 'currentRightBorder')
+        );
+        const tableBottomBorder: number = parseFloat(
+            vtable.table.getAttribute('currentBottomBorder')
+        );
+        const ratioX =
+            1.0 +
+            (this.isRTL
+                ? (currentBorder - mouseX) / (rect.right - currentBorder)
+                : (mouseX - currentBorder) / (currentBorder - rect.left));
+        const ratioY = 1.0 + (mouseY - tableBottomBorder) / (tableBottomBorder - rect.top);
+
+        const shouldResizeX = Math.abs(ratioX - 1.0) > 1e-3;
+        const shouldResizeY = Math.abs(ratioY - 1.0) > 1e-3;
+        if (shouldResizeX || shouldResizeY) {
+            for (let i = 0; i < vtable.cells.length; i++) {
+                for (let j = 0; j < vtable.cells[i].length; j++) {
+                    const cell = vtable.cells[i][j];
+                    if (cell.td) {
+                        if (shouldResizeX) {
+                            // the width of some external table is fixed, we need to make it resizeable
+                            vtable.table.removeAttribute('width');
+                            vtable.table.style.width = null;
+                            const originalWidth: number = cell.td.style.width
+                                ? parseFloat(
+                                      cell.td.style.width.substr(0, cell.td.style.width.length - 2)
```

Comment:
what does " - 2" mean here?
no magic number.

Reply:
It is used to convert string length to a number e.g. "19px" => 19, then we can calculate the new table width based on the ratio. I can add a constant for this.

Reply:
In that case I prefer use clientWidth instead of the value from CSS.

Reply:
fixed